src/d3a/models/market/one_sided.py

```diff
@@ -113,9 +113,12 @@ def accept_offer(self, offer_or_id: Union[str, Offer], buyer: str, *, energy: in
                         if self.area is None or self.area.bc is None \
                         else offer.real_id
 
+                    if trade_rate is None:
```

Comment:
It would also be nice to add an assert, which would validate whether the trade_rate respects the offer rate, since trade_rate should never fall below the initial offer rate.